File: 2260-minimum-consecutive-cards-to-pick-up/2260-minimum-consecutive-cards-to-pick-up.py
class Solution:
    def minimumCardPickup(self, cards: List[int]) -> int:
        
        n = len(cards)

        l = 0
        last_index = defaultdict(int)
        INF = 10 ** 10
        best = INF

        for r in range(n):
            x = cards[r]

            if x in last_index:
                cand = r - last_index[x] + 1
                if cand < best:
                    best = cand
            
            last_index[x] = r
        
        return best if best != INF else -1
            

File: 2260-minimum-consecutive-cards-to-pick-up/2260-minimum-consecutive-cards-to-pick-up.java
class Solution {
    public int minimumCardPickup(int[] cards) {
        Map<Integer, Integer> lastIndex = new HashMap<>();
        int n = cards.length;
        int best = Integer.MAX_VALUE;

        for (int i = 0; i < n; i++) {
            int val = cards[i];

            if (lastIndex.containsKey(val)) {
                int cand = i - lastIndex.get(val) + 1;
                best = Math.min(cand, best);
            }
            lastIndex.put(val, i);
        }

        return best == Integer.MAX_VALUE ? -1 : best;
    }
}